src/listing.rs

```diff
@@ -376,6 +376,15 @@ pub fn directory_listing(
                 .body(actix_web::body::BodyStream::new(rx)),
         ))
     } else {
+        if conf.disable_indexing {
+            return Ok(ServiceResponse::new(
+                req.clone(),
+                HttpResponse::NotFound()
+                    .content_type(mime::TEXT_PLAIN_UTF_8)
+                    .body("File not found."),
+            ));
+        }
+
```

Comment:
Doesn't putting it here result in the index still being downloadable via requesting an archive? This seems to me like a potential security problem or at least something people might not expect. I think it might be safer to prohibit index generation entirely in `main.rs`.

Reply:
I wasn't aware that was a thing. I'll take a look.

Reply:
After looking at it closer, I think it'll be more clean to just move this check all the way to the top of this function. All file listings seem to go through this handler.